PluralKit.Bot/Utils/AvatarUtils.cs

```diff
@@ -63,11 +64,24 @@ public static async Task VerifyAvatarOrThrow(HttpClient client, string url, bool
     // This lets us add resizing parameters to "borrow" their media proxy server to downsize the image
     // which in turn makes it more likely to be underneath the size limit!
     private static readonly Regex DiscordCdnUrl =
-        new(@"^https?://(?:cdn\.discordapp\.com|media\.discordapp\.net)/attachments/(\d{17,19})/(\d{17,19})/([^/\\&\?]+)\.(png|jpg|jpeg|webp)(\?.*)?$");
+        new(@"^https?://(?:cdn\.discordapp\.com|media\.discordapp\.net)/attachments/(\d{17,19})/(\d{17,19})/([^/\\&\?]+)\.(png|jpg|jpeg|webp)(?<query>\?.*)?$");
```

Comment:
i'd suggest changing the query capture here to not capture the question mark (something like `(?:\?(?<query>.*))?`) - makes rewriting the query string easier. 

with that, the `DiscordMediaUrlReplacement` string can stay the same as it was before this - and then change `TryRewriteCdnUrl` to something like this (untested, i'm writing this in the review comment field)

```csharp
public static string? TryRewriteCdnUrl(string? url) {
    if (url == null)
        return null;

    var match = DiscordCdnUrl.Match(url);
    var newUrl = DiscordCdnUrl.Replace(url, DiscordMediaUrlReplacement);
    if (match.Groups["query"].Success)
        newUrl += "&" + match.Groups["query"].Value;

    return newUrl;
}
```